### DISCUSSION TITLE
blog/updating-a-package-in-nixpkgs

### DISCUSSION THREAD

**Discussion by giscus** [JRMurr/JRMurr.github.io#2] [Comments]
# blog/updating-a-package-in-nixpkgs

My rough ramblings on how to contribute a small pr to Nixpkgs

https://johns.codes/blog/updating-a-package-in-nixpkgs

**Comment by dminca** [comment]
Interesting process, thanks for sharing!

On your last point with regards to _Handling feedback_: it's not a weird rule at all to always push 1 commit, they do it because `git` history tends to grow quite big in time and it makes the repo even more sluggish to operate (push/pull etc.)

If you ask me pushing these to git doesn't quite scale-up since you'll end up with a ton of commits from many contributors and sadly I don't have any other proposal.

**Reply by JRMurr** [reply]
Ahh fair point. For me im not as big of a fan since it makes re-reviews of a pr slightly harder to grok since its harder to tell what changed on the force push. 

I do understand the appeal for keeping the history clean, at my job for example we always do squash and merge on github to accomplish the same affect
